topics/command-arguments.md

```diff
@@ -0,0 +1,203 @@
+is the `arguments` field, which describes the arguments the command takes.
+
+It is a list, where every element is a map with information about the argument.
+
+These map fields are:
+ - `name`
+ - `type`
+ - `key-spec-index`
+ - `token`
+ - `summary`
+ - `since`
+ - `flags`
+ - `value`
+
+Only `name` and `type` are mandatory. At least one of `value` or `token` must exist.
+
+## name
+
+The name of the argument (for for identification, not what's displayed in the command syntax)
+
+## type
+
+The type of the argument
+
+The possible types are:
+ - `string`: String argument
+ - `integer`: Integer argument
+ - `double`: Double-precision argument
+ - `key`: String, but represents a key-name
+ - `pattern`: String, but represents a regex pattern
+ - `unix-time`: Integer, but represents a UNIX timestamp
+ - `pure-token`: The argument is just a token, which can exist or not (not a free-text input from user)
+ - `oneof`: The argument is a container of sub-arguments. Used when user can choose only one of a few sub-arguments (see example below)
+ - `block`: The argument is a container of sub-arguments. Used when one wants to group together several sub-arguments, usually to apply something onall of them (like making the entire group "optional") (see example below)
+
+### Example
+
+The trimming section of `XADD`, `[MAXLEN|MINID [=|~] threshold [LIMIT count]]` is a `block`, consisting of four sub-arguments:
+1. trimming startegy: this sub-argument is a `oneof`, with two sub-sub-arguments, each a `pure-token` (`MAXLEN` and `MINID`)
+2. trimming operator: this sub-argument is an optional `oneof`, with two sub-sub-arguments, each a `pure-token` (`=` and `~`)
+3. threshold: this sub-argument is a `string`
+4. count: this sub-argument is an optional `integer`, with a `token` (`LIMIT`)
+
+## key-spec-index
+
+If the command is of type `key` this fields must exist and it contains the index of the corresponding `key-spec` within the `key-specs` array, returned by `COMMAND`
+For more information please check the [key-specs page][tr].
+[tr]: /topics/key-specs
+
+## token
+
+Some arguments have a contant literal before the user input (unless it's a `pure-token`, in which case there's no free-text user input)
+
+## summary
+
+A short description of the argument
+
+## since
+
+The debut Redis version of this argument
+
+## flags
+
+Nested @array-reply of argument flags:
+
+Possible flags are:
+ - `optional`: Argument is optional (like `GET` in `SET` command)
+ - `multiple`: Argument may be repeated (like `key` in `DEL`)
+ - `multiple-token`: The argument may repeat itself, and so does its token (like `GET pattern` in `SORT`)
+
+## value
+
+Can be either:
+
+1. A string, in which case it is the placeholder of a user input. Used to be displayed when building the command syntax (along with `token`, if exists); or
+2. An array of maps, each representing a sub-argument (in case the `type` is `oneof` or `block`)
```

Comment:
```suggestion
The value can be one of the following:

1. A string, in which case it is the placeholder for the user's input. This is used when displaying the command's syntax (along with `token`, if it exists); or
2. In case the argument's `!type` is either `oneof` or `block`, the value is n array of maps, where each represents a nested argument
```